### PR TITLE
Fix issue #328 and some improvements.

### PR DESCRIPTION
**Pull request by Aceralon**
1.Fix issue 328 by returning error.
2.Remove named return for better code readability.
3.Rename variable name colliding with import name.

### CODE REVIEW

File: engine/virt/virt.go [hunk 1]
```diff
@@ -93,54 +93,52 @@ func (v *Virt) Execute(ctx context.Context, target string, config *enginetypes.E
 }
 
 // ExecExitCode gets return code of a specific execution.
-func (v *Virt) ExecExitCode(ctx context.Context, execID string) (code int, err error) {
-	return 0, nil
+func (v *Virt) ExecExitCode(ctx context.Context, execID string) (int, error) {
+	log.Warnf(ctx, "ExecExitCode does not implement")
+	return 0, nil //Called between other apis, cannot return error
```
Comment (CMGS) [inline engine/virt/virt.go]: space between // and notes

File: engine/virt/virt.go [hunk 2]
```diff
@@ -93,54 +93,52 @@ func (v *Virt) Execute(ctx context.Context, target string, config *enginetypes.E
...
+func (v *Virt) NetworkList(ctx context.Context, drivers []string) ([]*enginetypes.Network, error) {
+	return nil, fmt.Errorf("NetworkList does not implement")
 }
 
 // BuildRefs builds references, it's not necessary for virt. presently.
-func (v *Virt) BuildRefs(ctx context.Context, name string, tags []string) (refs []string) {
+func (v *Virt) BuildRefs(ctx context.Context, name string, tags []string) []string {
 	log.Warnf(ctx, "BuildRefs does not implement")
-	return
+	return nil //Called between other apis, cannot return error
```
Comment (CMGS) [inline engine/virt/virt.go]: space between // and notes

File: engine/virt/virt.go [hunk 3]
```diff
@@ -93,54 +93,52 @@ func (v *Virt) Execute(ctx context.Context, target string, config *enginetypes.E
...
-	return
+	return err
 }
 
 // NetworkList lists all of networks.
-func (v *Virt) NetworkList(ctx context.Context, drivers []string) (nets []*enginetypes.Network, err error) {
-	log.Warnf(ctx, "NetworkList does not implement")
-	return
+func (v *Virt) NetworkList(ctx context.Context, drivers []string) ([]*enginetypes.Network, error) {
+	return nil, fmt.Errorf("NetworkList does not implement")
```
Comment (CMGS) [inline engine/virt/virt.go]: can define a error first, in other place

File: engine/virt/virt.go [hunk 4]
```diff
@@ -211,7 +207,7 @@ func (v *Virt) VirtualizationInspect(ctx context.Context, ID string) (*enginetyp
 		return nil, err
 	}
 
-	bytes, err := json.Marshal(coretypes.LabelMeta{Publish: []string{"PORT"}})
+	marshal, err := json.Marshal(coretypes.LabelMeta{Publish: []string{"PORT"}})
```
Comment (CMGS) [inline engine/virt/virt.go:210]: why change this?

File: engine/virt/virt.go [hunk 5]
```diff
@@ -279,6 +275,6 @@ func (v *Virt) VirtualizationExecute(ctx context.Context, ID string, commands, e
 
 // ResourceValidate validate resource usage
 func (v *Virt) ResourceValidate(ctx context.Context, cpu float64, cpumap map[string]int64, memory, storage int64) error {
-	// TODO list all workloads, calcuate resource
-	return nil
+	// TODO list all workloads, calculate resource
+	return fmt.Errorf("VirtualizationExecute not implemented")
```
Comment (CMGS) [inline engine/virt/virt.go]: same here, you should define error first

File: engine/virt/virt.go (hunk 4 above)
Comment (Aceralon) [reply engine/virt/virt.go:210]: bytes collides with imported package bytes.

**Comment by CMGS** [comment]
check PR workflow, and make all pass

**Pull request merged**